lib/ansible/modules/system/cgroup.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Cédric Nisio <cedric.nisio@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: cgroup
+
+short_description: Manage cgroup configurations
+
+version_added: "2.6"
+
+description:
+    - Manages cgroup configurations in the /etc/cgconfig.d directory.
+    - Checks state using cgsnapshot.
+
+options:
+    name:
+        description:
+            - The cgroup name
+        required: true
+    state:
+        description:
+            - Whether the cgroup should be present or absent.
+        required: true
+        choices: ['absent', 'present']
+    memlimit:
+        description:
+            - The maximum RSS memory for the cgroup.
+        required: false
+    memswaplimit:
+        description:
+            - The maximum RSS memory and swap for the cgroup. This can't be lower than the memlimit.
+        required: false
+
+author:
+    - Cedric Nisio (@Dayde)
+'''
+
+EXAMPLES = '''
+# Checks a cgroup is present with only a memory limit
+- cgroup:
+    name: my_cgroup
+    state: present
+    memlimit: 256m
+
+# Checks a cgroup is absent
+- cgroup:
+    name: my_cgroup
+    state: absent
+'''
+
+RETURN = '''
+action:
+    description: If something changed the corresponding action (Creating, Updating or Deleting cgroup)
+    type: str
+    returned: only if changed
+config_diff:
+    description: If we're updating the configuration, the diff between the former and the new configuration
+    type: str
+    returned: only if updating configuration
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    module_args = dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', required=True),
```

Comment:
`state`: a `choice` parameter should be used ([example](https://github.com/ansible/ansible/blob/6cdf91fd35fefec35dca4594db231cc6af5a8620/lib/ansible/modules/system/capabilities.py#L167))

Reply:
Done